apps/testing/src/main/java/com/akto/testing/HTTPClientHandler.java

```diff
@@ -55,11 +55,10 @@ static class NormalResponseInterceptor implements Interceptor {
         @Override
         public @NotNull Response intercept(Chain chain) throws IOException {
             Request request = chain.request();
-
             try {
                 Buffer buffer = new Buffer();
                 RequestBody requestBody = request.body();
-                if (requestBody != null) {
+                if (requestBody != null && requestBody.contentLength() != -1) {
```

Comment:
can any other kind of request apart from file upload have requestBody.contentLength() != -1

Reply:
not in our case, because we have the body of type string which has fixed length